### PR TITLE
[SPARK-32683][DOCS][SQL] Fix doc error and add migration guide for datetime pattern F

### PR DESCRIPTION
**Pull request by yaooqinn**
### What changes were proposed in this pull request?

This PR fixes the doc error and add a migration guide for datetime pattern.

### Why are the changes needed?
This is a bug of the doc that we inherited from JDK https://bugs.openjdk.java.net/browse/JDK-8169482

The SimpleDateFormatter(**F Day of week in month**) we used in 2.x and the DatetimeFormatter(**F week-of-month**) we use now both have the opposite meanings to what they declared in the java docs. And unfortunately, this also leads to silent data change in Spark too.

The `week-of-month` is actually the pattern `W` in DatetimeFormatter, which is banned to use in Spark 3.x.

If we want to keep pattern `F`, we need to accept the behavior change with proper migration guide and fix the doc in Spark

### Does this PR introduce _any_ user-facing change?

Yes, doc changed

### How was this patch tested?

passing ci doc generating job

### CODE REVIEW

**Comment by yaooqinn** [comment]
cc @cloud-fan @maropu, thanks very very much~

File: docs/sql-migration-guide.md [hunk 1]
```diff
@@ -189,6 +189,69 @@ license: |
 
   - Since Spark 3.0, when using `EXTRACT` expression to extract the second field from date/timestamp values, the result will be a `DecimalType(8, 6)` value with 2 digits for second part, and 6 digits for the fractional part with microsecond precision. e.g. `extract(second from to_timestamp('2019-09-20 10:10:10.1'))` results `10.100000`.  In Spark version 2.4 and earlier, it returns an `IntegerType` value and the result for the former example is `10`.
 
+  - In Spark 3.0, datetime pattern letter `F` represents the concept of the count of days within the period of a week where the weeks are aligned to the start of the month, e.g.
+    ```sql
+    SELECT d, date_format(d, 'F') AS f
```
Comment (maropu) [inline docs/sql-migration-guide.md:194]: Could we move the example queries into the datetime doc, then add a link here to it? IMHO most users check the doc first, maybe...

Comment (cloud-fan) [reply docs/sql-migration-guide.md:194]: can we use one specific date value to demonstrate the difference?

Comment (yaooqinn) [reply docs/sql-migration-guide.md:194]: it may not seem intuitive enough with only one date.

Comment (yaooqinn) [reply docs/sql-migration-guide.md:194]: I removed some cases, maybe looks better now?

Comment (yaooqinn) [reply docs/sql-migration-guide.md:194]: > Could we move the example queries into the datetime doc, then add a link here to it? IMHO most users check the doc first, maybe...

SGTM, I will wait until the cases look ok to you guys. Or maybe we can just make a copy there.

Comment (maropu) [reply docs/sql-migration-guide.md:194]: Just coying them into the doc sounds okay to me.

Comment (cloud-fan) [reply docs/sql-migration-guide.md:194]: why one value isn't enough?
```
2020-07-30 is 4 weeks and 2 days after the first day of the month,
so `date_format(date'2020-07-30', 'F')` returns 2 in Spark 3.0 and 5 in Spark 2.x. 
```

File: docs/sql-migration-guide.md [hunk 2]
```diff
@@ -189,6 +189,49 @@ license: |
...
+    +------------+---+
+    | 2020-07-30 | 2 |
+    | 2020-07-31 | 3 |
+    | 2020-08-01 | 1 |
+    | 2020-08-07 | 7 |
+    | 2020-08-08 | 1 |
+    | 2020-08-09 | 2 |
+    +------------+---+
+    ```
+    In Spark version 2.4 and earlier, it is the week of month that represents the concept of the count of weeks within the month where weeks start on a fixed day-of-week, e.g.
```
Comment (cloud-fan) [inline docs/sql-migration-guide.md:213]: > where weeks start on a fixed day-of-week

what does it mean?

Comment (yaooqinn) [reply docs/sql-migration-guide.md:213]: FYI https://docs.oracle.com/javase/8/docs/api/java/time/temporal/WeekFields.html#weekOfMonth--

File: docs/sql-migration-guide.md (hunk 1 above)
Comment (yaooqinn) [reply docs/sql-migration-guide.md:194]: > why one value isn't enough?
> 
> ```
> 2020-07-30 is 4 weeks and 2 days after the first day of the month,
> so `date_format(date'2020-07-30', 'F')` returns 2 in Spark 3.0 and 5 in Spark 2.x. 
> ```

For 2.4, we can not say that, it's not related to the start day of month, but also need to consider the first day of week.

Comment (cloud-fan) [reply docs/sql-migration-guide.md:194]:
```
2020-07-30 is 30 days (4 weeks and 2 days) after the first day of the month, and Thursday of the fifth week of the month. so `date_format(date'2020-07-30', 'F')` returns 2 in Spark 3.0 and 5 in Spark 2.x.
```

We need to highlight the difference that, one is day count within the week, one is week count within the month.

**Comment by cloud-fan** [comment]
thanks, merging to master/3.0!

**Pull request closed without merging**